texastribune/donations
# Bad actor

#### What's this PR do?

Checks for bad actors. 

#### Why are we doing this? How does it help us?

To make it harder for bad actors to test stolen card numbers against our site.

#### How should this be manually tested?

- Use the review app: https://donations-testing-pr-585.herokuapp.com/ (also CNAMEd to https://bad-actor-test.texastribune.org) 
- Try one-time, recurring, Circle transactions. Try to make some of them trigger the quarantine (see wiki doc for what would do that).
- Check the #test-quarantine-donation Slack channel and approve or reject donations. Confirm that they end up that way in Salesforce.

#### How should this change be communicated to end users?

Let Sarah and anyone who might be approving transactions know and invite them to the approver channel. 

#### Are there any smells or added technical debt to note?

I don't think so.

#### What are the relevant tickets?

#### Have you done the following, if applicable:
***(optional: add explanation between parentheses)***

* [x] Updated the documentation/wiki? *(https://texastribune.atlassian.net/wiki/spaces/TECH/pages/907018241/Bad+Actors+card+testing?moved=true)*

#### TODOs / next steps:

Check the above wiki page.
